src/librustc/dep_graph/edges.rs

```diff
@@ -104,63 +160,52 @@ impl DepGraphEdges {
     /// you are not in a task; what is bad is *writing* to tracked
     /// state (and leaking data that you read into a tracked task).
     pub fn read(&mut self, v: DepNode) {
-        if self.current_node().is_some() {
-            let source = self.make_node(v);
-            self.add_edge_from_current_node(|current| (source, current))
+        match self.task_stack.last_mut() {
```

Comment:
Oh, I see what you mean. Indeed, this is broken. Ha, score one for my persnicketiness about being exhaustive. =)